tests/unittests/general/webfront_test.py

```diff
@@ -19,16 +19,16 @@ class TestLdapAuthenticate(object):
     def test_authenticate_should_return_account_when_ldap_says_yes(self):
         with patch("nav.web.ldapauth.available", new=True):
             with patch("nav.web.ldapauth.authenticate", return_value=True):
-                assert auth.authenticate('knight', 'shrubbery') == LDAP_ACCOUNT
+                assert auth.authenticate_ldap('knight', 'shrubbery') == LDAP_ACCOUNT
 
     def test_authenticate_should_return_false_when_ldap_says_no(self):
         with patch("nav.web.ldapauth.available", new=True):
             with patch("nav.web.ldapauth.authenticate", return_value=False):
-                assert not auth.authenticate('knight', 'shrubbery')
+                assert auth.authenticate_ldap('knight', 'shrubbery') is None
```

Comment:
Here you introduce a grueling difference between the name of the test and what the test actually does

Reply:
There's a difference between returning None and returning something False-y. This feels safer to me, I've been burnt too many times by an accidental False-y. Returning something False-y that is not None is in fact a bug in this case, and shouldn't ever happen, so I would very much like for a test to flag it.

Reply:
False or falsey, the point is that this is a test for the `authenticate` function, not the `authenticate_ldap` function. This teste that the former delegates to the latter and properly heeds its result.

Reply:
also, if you change the code so it should return None, rather than False, at least change the name of the test so it doesn't say it should return false and verifies that it returns None.

Reply:
Not the only test with a bad name here :) TestNormalAuthenticate.test_authenticate_should_return_false_when_ldap_says_no, when ldap is *off*. Ancient, messy code is messy, I guess.